app/src/main/java/com/arygm/quickfix/ui/dashboard/Dashboard.kt

```diff
@@ -0,0 +1,218 @@
+                          modifier = Modifier.padding(7.dp),
+                          contentPadding = PaddingValues(vertical = 0.dp, horizontal = 10.dp),
+                          border =
+                              if (quickFixFilterButtons[index].isSelected) null
+                              else BorderStroke(1.dp, colorScheme.primary),
+                      )
+                    }
+                  }
+                  when (quickFixFilterButtons.firstOrNull { it.isSelected }?.title) {
+                    "Upcoming" ->
+                        QuickFixesWidget(
+                            status = "Upcoming",
+                            quickFixList = quickFixes,
+                            onShowAllClick = { /* Handle Show All Click */},
+                            onItemClick = { /* Handle QuickFix Item Click */},
+                            modifier = Modifier.testTag("UpcomingQuickFixes"),
+                            itemsToShowDefault = 3,
+                        )
+                    "Canceled" -> {
+                      QuickFixesWidget(
+                          status = "Canceled",
+                          quickFixList = quickFixes,
+                          onShowAllClick = { /* Handle Show All Click */},
+                          onItemClick = { /* Handle QuickFix Item Click */},
+                          modifier = Modifier.testTag("UpcomingQuickFixes"),
+                          itemsToShowDefault = 3,
+                      )
+                    }
+                    "Unpaid" -> {
+                      QuickFixesWidget(
+                          status = "Unpaid",
+                          quickFixList = quickFixes,
+                          onShowAllClick = { /* Handle Show All Click */},
+                          onItemClick = { /* Handle QuickFix Item Click */},
+                          modifier = Modifier.testTag("UpcomingQuickFixes"),
+                          itemsToShowDefault = 3,
+                      )
+                    }
+                    "Finished" -> {
+                      QuickFixesWidget(
+                          status = "Finished",
+                          quickFixList = quickFixes,
+                          onShowAllClick = { /* Handle Show All Click */},
+                          onItemClick = { /* Handle QuickFix Item Click */},
+                          modifier = Modifier.testTag("UpcomingQuickFixes"),
+                          itemsToShowDefault = 3,
+                      )
+                    }
+                    else -> {
+                      QuickFixesWidget(
+                          quickFixList = quickFixes,
+                          onShowAllClick = { /* Handle Show All Click */},
+                          onItemClick = { /* Handle QuickFix Item Click */},
+                          modifier = Modifier.testTag("UpcomingQuickFixes"),
+                          itemsToShowDefault = 3,
+                      )
```

Comment:
It's good that you future proofed the UI to easily integrate the list of quickfixes depending on the selected filter 😄